crates/remote_server/src/unix.rs

```diff
@@ -692,11 +783,18 @@ fn spawn_server(paths: &ServerPaths) -> Result<()> {
 
     let status = server_process
         .status()
-        .context("failed to launch server process")?;
-    anyhow::ensure!(
-        status.success(),
-        "failed to launch and detach server process"
-    );
+        .map_err(SpawnServerError::ProcessStatus)?;
+
+    if !status.success() {
+        return Err(SpawnServerError::LaunchStatus {
+            status,
+            paths: format!(
+                "log file: {}, pid file: {}",
+                paths.log_file.display(),
```

Comment:
NIT: if we use `{:?}` instead of `.display()`, we'll get the paths quoted.
Might be better, given that certain ones might have whitespaces inside.